src/core/src/package_managers/YumPackageManager.py

```diff
@@ -845,6 +864,67 @@ def is_auto_update_service_installed(self, install_check_cmd):
+            self.composite_logger.log_debug("[YPM] Machine default auto OS update service is not installed on the VM and hence no config to revert. [Service={0}]".format(str(service)))
+            return
+
+        self.composite_logger.log_debug("[YPM] Logging current configuration settings for auto OS updates [Service={0}][Is_Service_Installed={1}][Machine_default_update_enable_on_reboot={2}][{3}={4}]][{5}={6}]"
+                                        .format(str(self.current_auto_os_update_service), str(is_service_installed), str(enable_on_reboot_value), str(self.download_updates_identifier_text), str(download_updates_value), str(self.apply_updates_identifier_text), str(apply_updates_value)))
+
+        image_default_patch_configuration_backup = self.__get_image_default_patch_configuration_backup()
+        self.composite_logger.log_debug("[YPM] Logging system default configuration settings for auto OS updates. [Settings={0}]".format(str(image_default_patch_configuration_backup)))
+        is_backup_valid = self.is_image_default_patch_configuration_backup_valid(image_default_patch_configuration_backup)
+
+        if is_backup_valid:
+            download_updates_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.download_updates_identifier_text]
+            apply_updates_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.apply_updates_identifier_text]
+            enable_on_reboot_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.enable_on_reboot_identifier_text]
+
+            self.update_os_patch_configuration_sub_setting(self.download_updates_identifier_text, download_updates_value_from_backup, self.auto_update_config_pattern_match_text)
+            self.update_os_patch_configuration_sub_setting(self.apply_updates_identifier_text, apply_updates_value_from_backup, self.auto_update_config_pattern_match_text)
+            if str(enable_on_reboot_value_from_backup).lower() == 'true':
+                self.enable_auto_update_on_reboot()
+        else:
+            self.composite_logger.log_debug("[YPM] Since the backup is invalid or does not exist for current service, we won't be able to revert auto OS patch settings to their system default value. [Service={0}]".format(str(service)))
+
+    def __init_auto_update_for_service(self, service):
+        """ Verifies if default auto update configurations, for a service under consideration, are saved in backup """
+        switcher = {
+            Constants.YumAutoOSUpdateServices.YUM_CRON: self.__init_auto_update_for_yum_cron,
+            Constants.YumAutoOSUpdateServices.DNF_AUTOMATIC: self.__init_auto_update_for_dnf_automatic,
+            Constants.YumAutoOSUpdateServices.PACKAGEKIT: self.__init_auto_update_for_packagekit
+        }
+        try:
+            return switcher[service]()
+        except KeyError as e:
+            raise e
```

Comment:
What is this doing? It seems like the code behavior with and without the try catch is all the same.

KeyError will be raised (explicitly). And non-KeyError will also get raised (implicitly).

Reply:
Followed pattern from other parts in code, 
eg: ActionHandler.py
<img width="216" alt="image" src="https://github.com/user-attachments/assets/25168b7a-9fac-482f-b216-eeef52aa5a9e" />

YumPackageManager.py
<img width="585" alt="image" src="https://github.com/user-attachments/assets/5cdba99b-e808-47d8-a6a3-8611777e1df1" />

Reply:
All that code does nothing useful either so wouldn't be a good precedent to take and replicate.

Reply:
Removed try catch